pkg/asset/machines/master.go

```diff
@@ -169,6 +172,13 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 				}
 			}
 		}
+		if mpool.InstanceType == "" {
+			mpool.InstanceType, err = aws.PreferedInstanceType(ctx, installConfig.AWS, awsDefaultMasterMachineTypes(installConfig.Config.Platform.AWS.Region), mpool.Zones)
+			if err != nil {
+				logrus.Warn(errors.Wrap(err, "failed to find default instance type"))
+				mpool.InstanceType = awsDefaultMasterMachineTypes(installConfig.Config.Platform.AWS.Region)[0]
```

Comment:
It's not clear to me why we want this to be a soft warning, instead of a fatal error.  Can we make it a fatal error, because a failure to find a default instance type can be worked around by users setting explicit types in their `install-config.yaml`?

Reply:
If the user doesn't have access to run this command, or it's not supported in all the regions or we are rate limited, i don't want the install to fail as this stage, _4.4 feature complete_

I think for now the current behavior of use the defaults when we can't dynamically verify it is good enough. we can revisit the requirement in next feature cycle.